pkg/roll/run_update.go

```diff
@@ -167,13 +167,44 @@ ROLL_LOOP:
 				break ROLL_LOOP
 			}
 
-			next := algorithm(oldNodes.Healthy, newNodes.Healthy, u.DesiredReplicas, u.MinimumReplicas)
+			next := rollAlgorithm(oldNodes.Healthy, newNodes.Healthy, u.DesiredReplicas, u.MinimumReplicas)
 			if next > 0 {
+				// apply the delay only if we've already added to the new RC, since there's
+				// no value in sitting around doing nothing before anything has happened.
+				if newNodes.Desired > 0 && u.RollDelay > time.Duration(0) {
+					u.logger.WithField("delay", u.RollDelay).Infof("Waiting %v before continuing deploy", u.RollDelay)
+
+					select {
+					case <-time.After(u.RollDelay):
+					case <-quit:
+						return
+					}
+
+					// Check health again following the roll delay. If things have gotten
+					// worse since we last looked, or there is an error, we break this iteration.
+					checks, err = u.hcheck.Service(newFields.Manifest.ID().String())
+					afterDelayNew, err := u.countHealthy(u.NewRC, checks)
+
+					if err != nil {
+						u.logger.WithError(err).Errorln("Could not count nodes after delay, skipping this iteration")
+						break
+					}
+
+					if afterDelayNew.Healthy < newNodes.Healthy {
+						u.logger.WithFields(logrus.Fields{
+							"beforeDelay": newNodes.Healthy,
+							"afterDelay":  afterDelayNew.Healthy,
+							"delay":       u.RollDelay,
+						}).Errorf("Healthy node count on the new RC dropped after %v roll delay, skipping this iteration", u.RollDelay)
+						break
+					}
+				}
```

Comment:
Do you think it would make sense to re-run `rollAlgorithm()` before continuing, in case the health has further changed.  I know, by this point, health is no worse than it was before waiting, but what if it's better?